lib/posix/syscall.h

```diff
@@ -0,0 +1,27 @@
+#ifndef LIB_POSIX_SYSCALL_H_
+#define LIB_POSIX_SYSCALL_H_
+
+#include <unistd.h>
+
+#include "lib/error/return_if_error.h"
+#include "lib/error/status_or.h"
+#include "lib/posix/errno.h"
+
+namespace posix {
+
+// Trivial syscall wrapper. Checks for error conditions and converts errno to
+// error::Status.
+//
+// Return the result of syscall() on success or an error status on failure.
+//
+// See 'man 2 syscall' for details.
+template <typename... ArgsT>
+inline error::StatusOr<int> Syscall(const int number, ArgsT&&... args) {
+  const auto rv = ::syscall(number, std::forward<ArgsT>(args)...);
+  RETURN_IF_ERROR(OkStatusOrCaptureErrnoIf(-1 == rv, "syscall() failed"));
```

Comment:
While we expect syscall() to return -1 or 0 exclusively, only the zero value is indisputably non-error. Should we rather test 0 != rv in the interests of being more robust?

Reply:
A number of syscalls return a file descriptor so 0 is actually less general. There is no single standard, but the wrapper above captures nearly all cases.

From man 2 syscall:
RETURN VALUE
       The return value is defined by the system call being invoked.  In general, a 0 return  value
       indicates  success.   A  -1  return value indicates an error, and an error code is stored in
       errno.

Reply:
Argh, please excuse my derp.